### ISSUE TITLE
TextField Acessibility ID not persist after build

### ISSUE THREAD

**Issue by paulowsky** [flutter/flutter#144357]
### Steps to reproduce

1. create a sample app
2. add a Button with Key property
3. add a TextField with Key property
4. build app
5. inspect built app using some tool like appium-inspector

### Expected results

Both Button and TextField expose accessibility-id.

### Actual results

Only Button shows accessility-id.

### Code sample

<details open><summary>Code sample</summary>

```dart
TextField(
  decoration: InputDecoration(
    border: OutlineInputBorder(),
    hintText: 'Enter a search term',
  ),
),

TextButton(
  style: ButtonStyle(
    foregroundColor: MaterialStateProperty.all<Color>(Colors.blue),
  ),
  onPressed: () { },
  child: Text('TextButton'),
)
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

<img width="1030" alt="image" src="https://github.com/flutter/flutter/assets/43378215/c1c76b4b-223d-4286-b99f-656408db9caf">
<img width="1006" alt="image" src="https://github.com/flutter/flutter/assets/43378215/f1877193-ed28-4a4f-b672-38d0b626d476">

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

Flutter 3.19 may have identifier for SemanticsProperties (introduced by flutter/flutter#138331). It sets resource-id and accessibilityIdentifier for Android and iOS, then UiAutomator2/XCUITest drivers also can handle Key without this driver
"appium:disableIdLocatorAutocompletion": true would be necessary to make resource-id idea work without any package name prefix like Android compose.
e.g. flutter/flutter#17988 (comment)

```console
[✓] Flutter (Channel stable, 3.19.0, on Ubuntu 23.10 6.5.0-21-generic, locale en_US.UTF-8)
    • Flutter version 3.19.0 on channel stable at /home/pedro/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision bae5e49bc2 (2 weeks ago), 2024-02-13 17:46:18 -0800
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /home/pedro/Development/Android/Sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /home/pedro/Development/Android/Sdk
    • Java binary at: /snap/android-studio/148/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.
[✓] Chrome - develop for the web
    • Chrome at google-chrome
[✓] Linux toolchain - develop for Linux desktop
    • Ubuntu clang version 16.0.6 (15)
    • cmake version 3.27.4
    • ninja version 1.11.1
    • pkg-config version 1.8.1
[✓] Android Studio (version 2023.1)
    • Android Studio at /snap/android-studio/148
    • Flutter plugin can be installed from:
      :hammer: https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      :hammer: https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
[✓] VS Code (version 1.86.2)
    • VS Code at /snap/code/current/usr/share/code
    • Flutter extension version 3.83.20240201
[✓] Connected device (3 available)
    • SM G770F (mobile) • RQ8NB02AVTF • android-arm64  • Android 13 (API 33)
    • Linux (desktop)   • linux       • linux-x64      • Ubuntu 23.10 6.5.0-21-generic
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 122.0.6261.69
[✓] Network resources
    • All expected network resources are available.
• No issues found!
```

</details>

**Comment by huycozy** [comment]
Thanks for the report. I also see the same on the latest master channel. This also happens on the previous release 3.16.9. 

<details>
<summary>Complete sample code</summary>

```dart
import 'package:flutter/material.dart';

void main() async {
  runApp(
    const MaterialApp(
      title: 'Flutter Example',
      home: MyApp(),
    ),
  );
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
				mainAxisAlignment: MainAxisAlignment.center,
        children: [
          const TextField(
            key: Key('search_field'),
            decoration: InputDecoration(
              border: OutlineInputBorder(),
              hintText: 'Enter a search term',
            ),
          ),
          TextButton(
            key: const Key('text_button'),
            style: ButtonStyle(
              foregroundColor: MaterialStateProperty.all<Color>(Colors.blue),
            ),
            onPressed: () {},
            child: const Text('TextButton'),
          ),
        ],
      ),
    );
  }
}

```
</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.19.2, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.19.2 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7482962148 (31 hours ago), 2024-02-27 16:51:22 -0500
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.86.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.82.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 122.0.6261.94

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.20.0-15.0.pre.45, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.20.0-15.0.pre.45 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2ccf375cb1 (40 minutes ago), 2024-02-28 23:17:36 -0500
    • Engine revision 232217f39c
    • Dart version 3.4.0 (build 3.4.0-187.0.dev)
    • DevTools version 2.33.0-dev.11
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.86.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.82.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 122.0.6261.94

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

**Comment by justinmc** [comment]
@hangyujin Are you the right person to confirm what the intended behavior is here for accessibility ids on Android?

**Comment by hangyujin** [comment]
I am not familiar with this, the feature was added at https://github.com/flutter/engine/pull/47961.

[bartekpacia](https://github.com/bartekpacia) Do you know what the intended behavior is here?

**Comment by bartekpacia** [comment]
Thanks for the ping @hangyujin. I believe author of this issue misunderstands the new "accessibility identifier" feature we added in Flutter 3.19. `key` on the widget doesn't have any effect, no changes were made to its behavior.

To add accessibility identifier to a widget, it has to be wrapped with `Semantics` and `Semantics.identifier` must be specified:

```dart
class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Semantics(
            identifier: 'search_field',
            child: TextField(
              decoration: InputDecoration(
                border: OutlineInputBorder(),
                hintText: 'Enter a search term',
              ),
            ),
          ),
          Semantics(
            identifier: Key('text_button')
            child: TextButton(
              style: ButtonStyle(
                foregroundColor: MaterialStateProperty.all<Color>(Colors.blue),
              ),
              onPressed: () {},
              child: const Text('TextButton'),
            ),
          ),
        ],
      ),
    );
  }
}
```

**Comment by justinmc** [comment]
Thanks @hangyujin @bartekpacia.

Does that fix your problem @paulowsky?
